Added Option to export csv between dates

ufoloko opened the pull request:
I have added an option to export to csv for data between two dates.

This works adding a timestamp field for UserDefinedRows table to be able to get records between two dates. This field is automatically populated when new row is added.

If **TWO** date fields are set in export section, results are filtered, otherwise all records will be exported.

Closes #100

WillStrohl commented on ExportCSV.ascx:
```diff
@@ -1,34 +1,58 @@
 <%@ Register TagPrefix="dnn" TagName="Label" Src="~/controls/LabelControl.ascx" %>
+<%@ Register TagPrefix="dnn" Namespace="DotNetNuke.Web.Client.ClientResourceManagement" Assembly="DotNetNuke.Web.Client" %>
+
+<dnn:DnnCssInclude runat="server" FilePath="~/Resources/Shared/components/TimePicker/Themes/jquery-ui.css" />
```
Are you using jQueryUI for something specific? Does it fix a related bug or something?

ufoloko replied:
Good day. Im using jqueryUI css for calendar visualization. DNN don't include styles just scripts for JqueryUI. If this file is not included calendar is showed without any style.

WillStrohl commented on ExportCSV.ascx:
```diff
@@ -1,34 +1,58 @@
+    </div>
+    <div class="dnnFormItem">
+        <dnn:label id="plInitialDate" runat="server" controlname="cbSystemFields" />
+        <asp:TextBox ID="txtInitialDate" CssClass="NormalTextBox" runat="server" MaxLength="200"
```
There's no validation to prevent errors and user experience issues. Also, the `readonly` attribute should probably be here and not only in the JavaScript, so that it's still `readonly` if JS is turned off - unless you have a technical reason for that. :)

WillStrohl replied:
@ufoloko So you're not able to validate that the value is in a date format or empty?  Maybe with a cominbation of Required, Custom, and Range validators to ensure the dates are in sequence and if one is entered, the other needs to be entered or deleted?

WillStrohl commented on ExportCSV.ascx:
```diff
@@ -1,34 +1,58 @@
+    </div>
+    <div class="dnnFormItem">
+        <dnn:label id="plFinalDate" runat="server" controlname="cbSystemFields" />
+        <asp:TextBox ID="txtFinalDate" CssClass="NormalTextBox" runat="server" MaxLength="200"
```
There's no validation to prevent errors and user experience issues. Also, the `readonly` attribute should probably be here and not only in the JavaScript, so that it's still `readonly` if JS is turned off - unless you have a technical reason for that. :)

WillStrohl commented on ExportCSV.ascx:
```diff
@@ -1,34 +1,58 @@
+    $(function () {
+      $("#<%= txtInitialDate.ClientID %>").attr("readonly", "readonly");
+      $("#<%= txtInitialDate.ClientID %>").datepicker({ dateFormat: "yy/mm/dd" });
+      $("#<%= txtFinalDate.ClientID %>").attr("readonly", "readonly");
```
This may or may not still be necessary, depending on how you want to handle the `readonly` attribute, as noted in another comment.  :)

ufoloko replied:
I think is explained on my other comment about to add readonly attribute from javascript

WillStrohl commented on ExportCSV.ascx.cs:
```diff
@@ -154,16 +155,40 @@ static string CleanName(string name)
-            var ds = new UserDefinedTableController(_moduleId, TabId, UserInfo).GetDataSet(true);
+            DataSet ds;
+
+            if(initialDate.Length>0 && finalDate.Length > 0)
```
This probably needs to be changed, but just as a note for performance for code in general, this might be better done using the built-in `string.IsNullOrEmpty()` method, because these are string values, and length could return a value greater than zero if there is one or more space values.

ufoloko replied:
Yes you are right. Sorry im not so experienced about how github works, how i can make this change to improve this PR.

WillStrohl replied:
No worries...  In the same branch you used to create the PR, just make the update and then push again as a new commit. This will update the PR for you.  :)  Nothing more is needed.

WillStrohl reviewed (changes requested):
Overall, this is a fantastic PR. Thank you for taking the time to make the contribution.  This is especially true since this module is still using a legacy development pattern.  Good stuff!  :) 

I left a few notes about some minor things, but these minor things could reduce errors later when this is in the next release.  

Again, well done.. The DNN community thanks you!  :)

valadas commented:
@ufoloko @WillStrohl if you want to get our head together to get this merged and do a new release, we are doing open-source co-coding events each Saturday on Discord, you are welcome to join us to get this PR pushed through the finish line and getting a new release out. https://discord.gg/m4RaJWd8

ufoloko commented:
Finally I have added suggestions you sent. About readonly feature, I need to add that option in client side because if you add that option from server, any change you make in dates is not being saved. I have added the option from server side but adding attributes to fields.

valadas reviewed (approved):
This looks good to me, I'll merge and we can iron out anything that crops up in other PRs.
